model/map/src/main/java/org/keycloak/models/map/user/MapUserProvider.java

```diff
@@ -118,26 +113,26 @@ private ModelException userDoesntExistException() {
         return new ModelException("Specified user doesn't exist.");
     }
 
-    private Optional<MapUserEntity<K>> getEntityById(RealmModel realm, String id) {
+    private Optional<MapUserEntity> getEntityById(RealmModel realm, String id) {
         try {
-            return getEntityById(realm, userStore.getKeyConvertor().fromString(id));
+            MapUserEntity mapUserEntity = tx.read(id);
+            if (mapUserEntity != null && entityRealmFilter(realm).test(mapUserEntity)) {
+                return Optional.of(mapUserEntity);
+            }
+
+            return Optional.empty();
         } catch (IllegalArgumentException ex) {
             return Optional.empty();
         }
     }
 
-    private MapUserEntity<K> getEntityByIdOrThrow(RealmModel realm, String id) {
+    private MapUserEntity getEntityByIdOrThrow(RealmModel realm, String id) {
         return getEntityById(realm, id)
                 .orElseThrow(this::userDoesntExistException);
     }
 
-    private Optional<MapUserEntity<K>> getEntityById(RealmModel realm, K id) {
-        MapUserEntity<K> mapUserEntity = tx.read(id);
-        if (mapUserEntity != null && entityRealmFilter(realm).test(mapUserEntity)) {
-            return Optional.of(mapUserEntity);
-        }
-
-        return Optional.empty();
+    private Optional<MapUserEntity> getRegisteredEntityById(RealmModel realm, String id) {
```

Comment:
Wrong rebase? this method should be removed.

Reply:
good catch!